services/frontend-service/src/legacy-ui/App/AuthContext.tsx

```diff
@@ -0,0 +1,126 @@
+import { ConfigsContext } from './index';
+import { Configuration, PublicClientApplication } from '@azure/msal-browser';
+import {
+    MsalProvider,
+    AuthenticatedTemplate,
+    useIsAuthenticated,
+    UnauthenticatedTemplate,
+    useMsal,
+} from '@azure/msal-react';
+
+type AuthContextType = {
+    useAzureAuth: boolean;
+    useAuth: boolean;
+};
+
+const AuthContext = React.createContext<AuthContextType>({} as AuthContextType);
+
+const getMsalConfig = (configs: any): Configuration => ({
+    auth: {
+        clientId: configs?.authConfig?.azureAuth?.clientId,
+        authority: `${configs?.authConfig?.azureAuth?.cloudInstance}${configs?.authConfig?.azureAuth?.tenantId}`,
+        redirectUri: configs?.authConfig?.azureAuth?.redirectURL,
+    },
+    cache: {
+        cacheLocation: 'sessionStorage',
+        storeAuthStateInCookie: false,
+    },
+});
+
+const getLoginRequest = () => ({
+    scopes: ['User.Read'],
+});
+
+type AuthTokenContextType = {
+    token: String;
+    authHeader: {
+        Authorization?: String;
+    };
+};
+
+export const AuthTokenContext = React.createContext<AuthTokenContextType>({} as AuthTokenContextType);
+
+function AzureAuthTokenProvider({ children }: { children: React.ReactNode }): JSX.Element {
+    const loginRequest = React.useMemo(() => getLoginRequest(), []);
+    const [token, setToken] = React.useState('');
+    const [authHeader, setAuthHeader] = React.useState({});
+    const { instance, accounts } = useMsal();
+
+    React.useEffect(() => {
+        const request = {
+            ...loginRequest,
+            account: accounts[0],
+        };
+        instance
+            .acquireTokenSilent(request)
+            .then((response) => {
+                setToken(response.accessToken);
+                setAuthHeader({ Authorization: response.accessToken });
+            })
+            .catch(() => {
+                instance.acquireTokenPopup(request).then((response) => {
+                    setToken(response.accessToken);
+                    setAuthHeader({ Authorization: response.accessToken });
+                });
+            });
+    }, [instance, accounts, loginRequest]);
+    return <AuthTokenContext.Provider value={{ token, authHeader }}>{children}</AuthTokenContext.Provider>;
+}
+
+const AzureAutoSignIn = () => {
+    const isAuthenticated = useIsAuthenticated();
+    const loginRequest = React.useMemo(() => getLoginRequest(), []);
+    const { instance } = useMsal();
+    if (!isAuthenticated) {
+        instance.loginRedirect(loginRequest);
+    }
+    return <></>;
+};
+
+export function AuthProvider({ children }: { children: React.ReactNode }): JSX.Element {
+    const { configs } = React.useContext(ConfigsContext);
+    const useAzureAuth = configs?.authConfig?.azureAuth?.enabled || false;
+    const useAuth = useAzureAuth;
+    const msalConfig = React.useMemo(() => getMsalConfig(configs), [configs]);
+    const msalInstance = new PublicClientApplication(msalConfig);
```

Comment:
Is It ok to send ` auth: {
        clientId: null,
        authority:  null,
        redirectUri: null,
    },
    cache: {
        cacheLocation: 'sessionStorage',
        storeAuthStateInCookie: false,
    },` to `new PublicClientApplication()` 
 Please add some tests for `AuthProvider`

Reply:
How would you suggest adding tests for AuthProvider. All the data given to authprovider comes from other contexts, and it uses azure components and functions to function properly.

Reply:
I answered you before 
`
You can import the contexts with mocked values, I'm not sure how @azure/msal-react works but I guess it should redirect you or open a popup window to authenticate, so the test should make sure if you give the right input @azure/msal-react should display the right output
`
`@azure/msal-react` is a third party library we can't just say ` it uses azure components and functions to function properly` we can't trust them blindly, it's better to have some tests to simulate all possible use cases.

Reply:
for 1, No. the function doesn't accept empty values its given by microsoft and dont have access to it.  So updated the structure so that the function will be called only when azure auth is enabled.

Reply:
even for testing msalprovider, you'd need an azure ad app client id.

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/test/TestConstants.ts 
https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/test/components/AuthenticatedTemplate.spec.tsx

So would still be dependant on someone elses azure ad app, which isn't ideal either.

Reply:
added tests